File: src/main/java/org/cloudfoundry/credhub/view/UserView.java
package org.cloudfoundry.credhub.view;

import org.cloudfoundry.credhub.credential.UserCredentialValue;
import org.cloudfoundry.credhub.domain.UserCredentialVersion;

public class UserView extends CredentialView {
  public UserView(UserCredentialVersion userCredential) {
    super(
        userCredential.getVersionCreatedAt(),
        userCredential.getUuid(),
        userCredential.getName(),
        userCredential.getCredentialType(),
        new UserCredentialValue(userCredential.getUsername(), userCredential.getPassword(), userCredential.getSalt())
    );
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/views/UserView.kt
package org.cloudfoundry.credhub.views

import org.cloudfoundry.credhub.credential.UserCredentialValue
import org.cloudfoundry.credhub.domain.UserCredentialVersion

class UserView(
    userCredential: UserCredentialVersion,
) : CredentialView(
        userCredential.versionCreatedAt,
        userCredential.uuid,
        userCredential.name,
        userCredential.getCredentialType(),
        userCredential.metadata,
        UserCredentialValue(userCredential.username, userCredential.password, userCredential.salt),
    )
